tests/app/authentication/test_jwe.py

```diff
@@ -0,0 +1,172 @@
+import unittest
+from tests.app.authentication import RRM_PUBLIC_PEM, SR_PRIVATE_PEM
+from tests.app.authentication.encoder import Encoder
+from app.authentication.jwt_decoder import Decoder
+from app.authentication.invalid_token_exception import InvalidTokenException
+import os
+
+valid_signed_jwt = "eyJraWQiOiJFRENSUk0iLCJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJleHAiOiIyMDc1Mjk3MTQ4IiwiaWF0IjoiMTQ1N" \
+                   "DkzNzAyMyIsInVzZXIiOiJqaW1teSJ9.gXARjhMIgthJahF60YHD4KEMifLw6TX8m6YpoYWekXqr76Sw86pyb-qA3R-0fWqqaq" \
+                   "RNAgx4au9t4uk9P5f3XMIXayPWpXhv7Iqqae7K-LIiRYz_7Jvim9fgO5JAsAp1qRWYkw2J92qhO87OUTacVLIXibHT4G2-W7Zr" \
+                   "5bHlzr3E4uJSTxJ1uYm3sCyYSh17MTvE8whXYkXny33SKVqKT2HY2EAEhIaDAZJ7ewjTMxCpwyHWGnJg5NFuaPY_CyZeZFNE17" \
+                   "rRlin2ThdydAWeyEpm5ffuH3uLhR6LyxTyyOP9AyCRBP50VR6xwJ3xYEmQis-4CCS3gKD-aAUn2yGl_g"
+
+valid_jwe = "eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAifQ.YfYdaxwHqcHrAUy3cMlOglZqGEv48bEfPse1-Pd8DU" \
+            "fo5Kg_nWEoP8NgCzfWP862dr6B7UAUUlbs-JLFmua4OlXBHCJG4IcQMxzJUjkoGxQIli8N59kwynyqdVvf22j5zggMokKIwACidPBzgu7" \
+            "aHdRq2PvtpjkZRE0Ga4Q6iF4BDSwtsI0zXn0nQsXXfTpz_uIlwPnCLUgMydg478U5t9qe24TsHfq-mimCFQ0HonH0b086VHoTESwTuiJ_" \
+            "0Kv9S-JiflZHmdAt_Xx0V0n7Lo9q97Q64gVWW8OKDrvLFQyMaUenFo0ONWpbtOgJlSm-xWrG7SIrm-uDMVeYm5lI2A.SNq_54mmzqYjp0" \
+            "y7.sHbanOmfxBsO-JOS-6nZ0HsgouMWary34i0Czx9VjCiB7tNig7nEib8nIHqRSs2ljubEhATGJXiptumZ4RoBHJSUQrp_vZBeR_EnKF" \
+            "XE8e45jz4CfBmuQo2W5lJoj2QvaVPBNrbCUsN62NIKb8n1Tr9JL0fnA6IQIByc8zR3TbrmnDYVEtgbY4IPRDsFI14Dm2cWaTYkj3c0A_4" \
+            "vPAhHsZEiGLQiKsy8WSoi8NC-CNs6FTqYrqcB4V77umi5XRU_JcdE_mqxcMkIHYUsUBDCqSMSA8xQgD56OOo41K4-vFMh99fU1u5zSbdq" \
+            "hqLtNeYTepJkOnCFi5aSjFPrtpfbvk9YPDJERJzEXAJkYbkUplmNudtObZY2IDmZglC5waGsTOq62tpudGPNb5B_FEq0s8ezAtMtf-C3h" \
+            "BslOcCLtAlGdRsptS6-L4YHxaGEAvwXD1heQADoAabjGCA-BZGPdc6h8A6y1QR7m2P33KEz83h141X4Ftyd9kTGxgjQ8-UrIGvu9b1wbm" \
+            "cjTWK3hiiyC3o0guBbzBq0SflX-RG9_zyK4sAIksiXj9mNRah1IUjUT5zr9cs1c5i4WwJdpJ28IzKvczgQ63B_t1xEpGsb6A35EhxzRkG" \
+            "7BQ.I9WExvUvZVXdwjxz8EPtCg"
+
+
+class JWETest(unittest.TestCase):
+    def test_valid_jwe(self):
+        decoder = Decoder(RRM_PUBLIC_PEM, SR_PRIVATE_PEM, "digitaleq")
```

Comment:
Consider moving the password into the test.app.authentication.__init__.py as well?